WL-3615 : Roster tool doesn't seem to like our sections

nicholaswilson100 opened the pull request:
Bug was code throwing IdNotFoundException when trying to find a section in coursemanagement which it might not have.
Surrounding it with empty try catch in roster2 to catch this case.

nicholaswilson100 committed:
WL-3615 : Roster tool doesn't seem to like our sections

Code throwing IdNotFoundException when trying to find a section in coursemanagement which it might not have.
Surrounding it with empty try catch in roster2 to catch this case.

buckett commented:
There look to be several other calls to the course management service from the roster2 code, do these calls not throw exceptions?

nicholaswilson100 commented:
There are a couple of other methods in RosterSiteEntityProvider. 

getMembership is called from getSite (which is the one we fixed).
getEnrollment  throws the same exception as getSite on the SakaiProxyImpl line 599.  I've put a similar try catch round that.  It's the only use of getEnrollmentSet in that submodule.   

RosterPOIEntityProvider (used for exporting to Excel) also uses code from getSite but since we've fixed that that's okay.)

nicholaswilson100 committed:
WL-3615 : try catch for another method in RosterSiteEntityProvider

private List<RosterEnrollment> getEnrollmentSets(String siteId,
			GroupProvider groupProvider)
also throws an exception
